pkg/sbom/v1/sbom_spdx_storage_format.go

```diff
@@ -50,7 +50,9 @@ func (sbom *SBOMData) StoreSBOM(sbomData any) error {
 	for i := range sbom.spdxData.Spec.SPDX.Files {
```

Comment:
If any of the inner files are empty, you should quit and stop sniffing.
But make sure you are not accessing here pointers without checking if they are nil.

Reply:
I don't think that there is a SBOM with no file in it.